test/server/cards/02_SHD/units/GentleGiant.spec.ts

```diff
@@ -0,0 +1,87 @@
+describe('Gentle Giant', function() {
+    integration(function(contextRef) {
+        describe('Gentle Giant\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [
+                            { card: 'gentle-giant', damage: 1 },
+                            { card: 'r2d2#ignoring-protocol', damage: 3 },
+                            { card: 'c3po#protocol-droid', damage: 1 }],
+                    },
+                    player2: {
+                        groundArena: [{ card: 'wampa', damage: 1 }]
+                    }
+                });
+            });
+
+            it('should heal a target with 1 damage to full', function () {
+                const { context } = contextRef;
+
+                // Attack
+                context.player1.clickCard(context.gentleGiant);
+                expect(context.gentleGiant).toBeInLocation('ground arena');
+                expect(context.player1).toBeAbleToSelectExactly([context.p2Base, context.wampa]);
+                context.player1.clickCard(context.p2Base);
+
+                // Healing Target
+                expect(context.player1).toBeAbleToSelectExactly([context.r2d2, context.c3po, context.wampa]);
+                context.player1.clickCard(context.c3po);
+
+                // Confirm Results
+                expect(context.gentleGiant.exhausted).toBe(true);
+                expect(context.c3po.damage).toBe(0);
+            });
+
+            it('should heal 1 damage from a unit', function () {
```

Comment:
Typically we aggregate multiple test cases into one if easily possible. This is bad test practice but we have a practical problem where the test framework we use is very slow and heavy and we have limited free github actions time in the free tier. So we need to consolidate a bit.

Typically with on-attack abilities like this we'll chain them together with a reset function that un-exhausts and heals relevant units, with comments for each case. Take a look at the tests for Rukh as an example of how we typically set things up.